core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -287,6 +294,69 @@ public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespac
   public void close() throws IOException {
     namespaces.clear();
     tables.clear();
+    views.clear();
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    if (!namespaceExists(namespace) && !namespace.isEmpty()) {
+      throw new NoSuchNamespaceException(
+          "Cannot list views for namespace. Namespace does not exist: %s", namespace);
+    }
+
+    return views.keySet().stream()
+        .filter(t -> namespace.isEmpty() || t.namespace().equals(namespace))
+        .sorted(Comparator.comparing(TableIdentifier::toString))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public View loadView(TableIdentifier identifier) {
+    if (isValidIdentifier(identifier) && views.containsKey(identifier)) {
+      return new BaseView(newViewOps(identifier), fullTableName(name(), identifier));
+    }
+
+    throw new NoSuchViewException("Invalid view identifier: %s", identifier);
```

Comment:
This isn't necessarily true. If the identifier is invalid then this should be thrown. If the view doesn't exist, this should be `View does not exist: %s`.